tests/test_serialization.py

```diff
@@ -137,6 +150,44 @@ def test_pickle_list() -> None:
         fp.seek(0)
         c1 = pickle.load(fp)
         assert c == c1
+        assert get_ref_type(c1) == Optional[List[Any]]
+        assert c1._metadata.element_type is None
+        assert c1._metadata.optional is True
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "input_,node,element_type,key_type,optional,output",
+    [
+        (GenericList, "list", None, None, False, List[Any]),
+        (OptGenericList, "opt_list", None, None, True, List[Any]),
+        (GenericDict, "dict", None, None, False, Dict[Union[str, Enum], Any]),
+        (OptGenericDict, "opt_dict", None, None, True, Dict[Union[str, Enum], Any]),
+        (SubscriptedDict, "dict", int, str, False, Dict[str, int]),
+        (SubscriptedList, "list", int, None, False, List[int]),
```

Comment:
Add some regular ListConfig and DictConfig with specified key and element types.